pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -126,6 +132,62 @@ def expval(self, measurementprocess: MeasurementProcess):
             measurementprocess.obs.name, measurementprocess.obs.wires
         )
 
+    # pylint: disable=protected-access
+    def probs(self, measurementprocess: MeasurementProcess):
+        """Probabilities of the supplied observable or wires contained in the MeasurementProcess.
+
+        Args:
+            measurementprocess (StateMeasurement): measurement to apply to the state
+
+        Returns:
+            Probabilities of the supplied observable or wires
+        """
+        diagonalizing_gates = measurementprocess.diagonalizing_gates()
+        if diagonalizing_gates:
+            self._qubit_state.apply_operations(diagonalizing_gates)
+        results = self._measurement_lightning.probs(measurementprocess.wires.tolist())
+        if diagonalizing_gates:
+            self._qubit_state.apply_operations(
+                [qml.adjoint(g) for g in reversed(diagonalizing_gates)]
+            )
+        return results
+
+    # pylint: disable=protected-access
+    def var(self, measurementprocess: MeasurementProcess):
```

Comment:
Not related to the PR, general lightning question: I noticed that neither `expval` nor `var` use diagonalizing gates in the measurement pipeline. Does lightning always use the observable matrix for an inner product?

Reply:
Yes, this is how it is implemented in C++.